typescript/sdk/src/fee/EvmTokenFeeDeployer.ts

```diff
@@ -79,7 +79,12 @@ export class EvmTokenFeeDeployer extends HyperlaneDeployer<
     ReturnType<EvmTokenFeeFactories[TokenFeeConfig['type']]['deploy']>
   > {
     let { maxFee, halfAmount } = config;
-    if (config.type === TokenFeeType.LinearFee && config.bps) {
+    if (
+      config.type === TokenFeeType.LinearFee &&
+      config.bps &&
+      !maxFee &&
+      !halfAmount
+    ) {
```

Comment:
shouldn't this check be:
```typescript
config.type === TokenFeeType.LinearFee &&
      config.bps &&
      (!maxFee ||
      !halfAmount)
```

as if maxFee or halfAmount are undefined the first branch of the if statement won't run but it would be an error running the second one as one of the 2 values is undefined and deployment might fail no?

Reply:
Fixed. good catch